samples/openapi3/client/petstore/php/OpenAPIClient-php/lib/Model/Animal.php

```diff
@@ -186,7 +186,7 @@ public function __construct(array $data = null)
         $this->container['color'] = isset($data['color']) ? $data['color'] : 'red';
 
         // Initialize discriminator property with the model name.
-        $discriminator = array_search('className', self::$attributeMap, true);
+        $discriminator = array_search('class_name', self::$attributeMap, true);
         $this->container[$discriminator] = static::$openAPIModelName;
```

Comment:
https://travis-ci.org/OpenAPITools/openapi-generator/builds/471525982?utm_source=github_status&utm_medium=notification

```
1) OpenAPI\Client\ModelInheritanceTest::testDiscriminatorInitialization
Failed asserting that null is identical to 'Dog'.
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/php/OpenAPIClient-php/tests/ModelInheritanceTest.php:70
```

---


```diff
+       $discriminator = array_search('class_name', self::$attributeMap, true);
        $this->container[$discriminator] = static::$openAPIModelName;	
```

Due to this PR, `array_search` may become unnecessary. We can pass `{{discriminatorName}}` to `$this->container` as following.

- Mustache template

`$this->container[{{discriminatorName}}] = static::$openAPIModelName;`

- PHP code

```diff
-       $discriminator = array_search('class_name', self::$attributeMap, true);
+       $this->container['class_name'] = static::$openAPIModelName;	
```

Reply:
👌 I'll make the change later tonight.